uint32 id_
float32 rate_
